src/backend/distributed/executor/multi_executor.c

```diff
@@ -45,12 +45,124 @@
  * TRUNCATE and real-time SELECT queries.
  */
 int MultiShardConnectionType = PARALLEL_CONNECTION;
+bool WritableStandbyCoordinator = false;
 
 
 /* local function forward declarations */
+static bool IsCitusPlan(Plan *plan);
+static bool IsCitusCustomScan(Plan *plan);
 static Relation StubRelation(TupleDesc tupleDescriptor);
 
 
+/*
+ * CitusExecutorStart is the ExecutorStart_hook that gets called when
+ * Postgres prepares for execution or EXPLAIN.
+ */
+void
+CitusExecutorStart(QueryDesc *queryDesc, int eflags)
+{
+	PlannedStmt *plannedStmt = queryDesc->plannedstmt;
+
+	/*
+	 * We cannot modify XactReadOnly on Windows because it is not
+	 * declared with PGDLLIMPORT.
+	 */
+#ifndef WIN32
+	if (RecoveryInProgress() && WritableStandbyCoordinator &&
+		IsCitusPlan(plannedStmt->planTree))
+	{
+		PG_TRY();
+		{
+			/*
+			 * To enable writes from a hot standby we cheat our way through
+			 * the checks in standard_ExecutorStart by temporarily setting
+			 * XactReadOnly to false.
+			 */
+			XactReadOnly = false;
+			standard_ExecutorStart(queryDesc, eflags | EXEC_FLAG_SKIP_TRIGGERS);
```

Comment:
What is `EXEC_FLAG_SKIP_TRIGGERS` for? Maybe add a comment.

Reply:
It seemed logical to skip triggers, but there's actually no risk in allowing them and letting them fail if they try to write. It's kind of a weird esoteric case anyway. Removing the flag.